### PR TITLE
Feature/Bugfix: add setgroups and getgroups to Credential functions

### PR DESCRIPTION
**Pull request by daveol**
This commit removes the complaints of rpmlint while pacakging ola for fedora,
but also makes the api more complete.

### CODE REVIEW

**Comment by daveol** [comment]
This pull request currently:
- Fixes #985 
- Has no unit-tests (yet)

I'm also not too sure about the implementation of [GetGroups()](https://github.com/OpenLightingProject/ola/pull/987/files#diff-cef0965a77aa8509f5ce9be919653b2aR149)

File: include/ola/base/Credentials.h [hunk 1]
```diff
@@ -120,6 +120,32 @@ bool SetGID(gid_t new_gid);
...
+ *
+ * @name Get / Set supplementary group IDs
+ * @{
+ */
+
+
+/**
+ * @brief Get the supplementary group ID's of the process
+ * @param size the size of the list to place group ID's in
+ * @param list the list to place group ID's
```
Comment (peternewman) [inline include/ola/base/Credentials.h]: [out]

**Comment by peternewman** [comment]
Just one comment from me.

**Comment by nomis52** [comment]
LGTM

**Comment by peternewman** [comment]
LGTM.

If you're feeling keen @daveol do you fancy fixing the other [out]s that are missing from the existing code in that file?

**Comment by nomis52** [comment]
LGTM

**Pull request merged**